### PR TITLE
[games101-cgl] new port

### PR DESCRIPTION
**Pull request by endingly**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [x] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html)
- [x] The versioning scheme in `vcpkg.json` matches what upstream says.
- [x] The license declaration in `vcpkg.json` matches what upstream says.
- [x] The installed as the "copyright" file matches what upstream says.
- [x] The source code of the component installed comes from an authoritative source.
- [x] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is in the new port's versions file.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by endingly** [comment]
@microsoft-github-policy-service agree

**Comment by endingly** [comment]
@Cheney-W Please wait for a few days. There are some issues with this library now.

**Comment by Cheney-W** [comment]
Moving the PR to draft. Please set it to "ready for review" once issues have been solved.

**Comment by endingly** [comment]
@Cheney-W Ready to review

File: ports/games101-cgl/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+vcpkg_copy_pdbs()
+
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/share")
+
+file(
+    INSTALL "${SOURCE_PATH}/license"
+    DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}"
+    RENAME copyright
+)
```
Comment (Cheney-W) [inline ports/games101-cgl/portfile.cmake:25]:
```suggestion
vcpkg_install_copyright(FILE_LIST "${SOURCE_PATH}/license")
```

File: versions/baseline.json [hunk 2]
```diff
@@ -8957,4 +8961,4 @@
       "port-version": 3
     }
   }
-}
+}
```
Comment (Cheney-W) [inline versions/baseline.json:8964]: Please don't modify this file manually.

**Comment by Cheney-W** [comment]
Could you please release a tag from https://github.com/endingly/games101-cgl and use the tag instead of the commit id?

File: versions/baseline.json (hunk 2 above)
Comment (endingly) [reply versions/baseline.json:8964]: So now do I need to delete? or just like that?

File: ports/games101-cgl/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    SOURCE_PATH "${SOURCE_PATH}"
+)
+
+vcpkg_cmake_install()
+vcpkg_copy_pdbs()
+
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/share")
+
+vcpkg_install_copyright(FILE_LIST "${SOURCE_PATH}/license")
```
Comment (Cheney-W) [inline ports/games101-cgl/portfile.cmake:19]:
```suggestion
vcpkg_install_copyright(FILE_LIST "${SOURCE_PATH}/license")

```

File: ports/games101-cgl/vcpkg.json [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+  "homepage": "https://github.com/endingly/games101-cgl",
+  "license": "MIT",
+  "dependencies": [
+    "freetype",
+    "glew",
+    "glfw3",
+    {
+      "name": "vcpkg-cmake",
+      "host": true
+    }
```
Comment (Cheney-W) [inline ports/games101-cgl/vcpkg.json:15]:
```suggestion
    },
    {
      "name": "vcpkg-cmake-config",
      "host": true
    }
```

File: ports/games101-cgl/portfile.cmake [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+    HEAD_REF main
+)
+
+vcpkg_cmake_configure(
+    SOURCE_PATH "${SOURCE_PATH}"
+)
+
+vcpkg_cmake_install()
+vcpkg_copy_pdbs()
+
```
Comment (Cheney-W) [inline ports/games101-cgl/portfile.cmake:15]:
```suggestion
vcpkg_cmake_config_fixup()

```

**Comment by Cheney-W** [comment]
Moving the PR to draft. Please set it to "ready for review" once changes have been applied.

**Review by Cheney-W — APPROVED** [review]

**Comment by Cheney-W** [comment]
Usage passed with following triplets:
```
x86-windows 
x64-windows 
x64-windows-static 
```

**Pull request merged**